xds/src/main/java/io/grpc/xds/GcpAuthenticationFilter.java

```diff
@@ -59,15 +59,17 @@ final class GcpAuthenticationFilter implements Filter {
 
   static final String TYPE_URL =
       "type.googleapis.com/envoy.extensions.filters.http.gcp_authn.v3.GcpAuthnFilterConfig";
-
+  private final LruCache<String, CallCredentials> callCredentialsCache;
   final String filterInstanceName;
 
-  GcpAuthenticationFilter(String name) {
+  GcpAuthenticationFilter(String name, int cacheSize) {
     filterInstanceName = checkNotNull(name, "name");
+    this.callCredentialsCache = new LruCache<>(cacheSize);
   }
 
-
   static final class Provider implements Filter.Provider {
+    long cacheSize = 10;
```

Comment:
Why are you making a `long` constant to always cast it to `int`?

Oh, you've deleted the default in `parseFilterConfig()`. No, no. `parseFilterConfig()` should not have any side-effects. The default is 10, not "whatever the last cache size was."